### ISSUE TITLE
Incorrect warning for empty lambda bodies with google_checks.xml

### ISSUE THREAD

**Issue by cushon** [checkstyle/checkstyle#6381]
```java
class T {
  Runnable runnable = () -> {};
}
```

```
$ java -jar checkstyle-8.16-all.jar -c google_checks.xml T.java
Starting audit...
[WARN] /tmp/T.java:2:29: WhitespaceAround: '{' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] /tmp/T.java:2:30: WhitespaceAround: '}' is not preceded with whitespace. [WhitespaceAround]
Audit done.
```

I used `google_checks.xml` from: https://raw.githubusercontent.com/checkstyle/checkstyle/master/src/main/resources/google_checks.xml

Both warnings are incorrect. 

The first warning cites [§4.1.3](https://google.github.io/styleguide/javaguide.html#s4.1.3-braces-empty-blocks), but the style guide clarifies that 'multi-block statement' refers to `if/else` or `try/catch/finally`. An empty lambda like `() -> {}` doesn't contain multiple blocks (and also isn't a statement).

The style guide provides a complete list of places where horizontal whitespace is required in [§4.6.2](https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace) and does not mention lambda bodies.

**Comment by rnveach** [comment]
It seems that this was added with #2924 when the token was added to the default tokens as google XML config uses the default tokens. I am seeing no conversations about google config in the issue or PR, so it could have been by accident.

We have had issues with the style guide as it isn't always complete and is sometimes ambiguous and google has never responded to issues we have raised with their wording.

> §4.1.3, but the style guide clarifies that 'multi-block statement' refers to if/else or try/catch/finally. An empty lambda like () -> {} doesn't contain multiple blocks

@romani I agree with the reading of the guide here.
I also think we should remove all default token usage in google's config and hardcode the list of tokens it uses. This way, if a new default token is added to a check then `AllChecksTest.testAllCheckTokensAreReferencedInGoogleConfigFile` should ping it as not being used for google and force us to review it.

If you agree with everything, please approve the issue.

**Comment by romani** [comment]
as @rnveach stated, workaround is to update `tokens` and remove LAMBDA from it. Full list of tokens is at https://checkstyle.org/config_whitespace.html#WhitespaceAround_Properties

 >  should ping it as not being used for google and force us to review it

good idea. But should be register in separate issue.

**Comment by ehiggs** [comment]
Related: https://github.com/google/google-java-format/issues/334

**Comment by romani** [comment]
Fix is merged.
@fzdy1914, thanks a lot for your contribution!
